src/public/Convert-SentinelARArmToYaml.ps1

```diff
@@ -140,106 +184,158 @@ function Convert-SentinelARArmToYaml {
+                    # Use original filename as new filename
+                    if ($resourceCounter -eq 0) {
+                        $NewFileName = $FileObject.Name -replace $FileObject.Extension, ".yaml"
+                    } else {
+                        $NewFileName = $FileObject.BaseName + "_$resourceCounter" + ".yaml"
+                    }
+                }
+                if ($UseDisplayNameAsFilename) {
+                    # Use the display name of the Analytics Rule as filename
+                    $NewFileName = $resource.properties.displayName -Replace '[^0-9A-Z]', ' '
+                    # Convert To CamelCase
+                    $NewFileName = ((Get-Culture).TextInfo.ToTitleCase($NewFileName) -Replace ' ') + '.yaml'
+                }
+                if ($UseIdAsFilename) {
+                    # Use id as of the Analytics Rule filename
+                    $NewFileName = $Id + '.yaml'
+                }
+
+                $OutFile = Join-Path $FileObject.Directory $NewFileName
+            } elseif ($PsCmdlet.ParameterSetName -in ("Path") -or ($PsCmdlet.ParameterSetName -in ("Pipeline") -and $OutFile -and -not $Directory)) {
+                if ($resourceCounter -gt 0) {
+                    $OutFile = $OutFile -replace ".yaml", "_$resourceCounter.yaml"
+                }
+            } elseif ($PsCmdlet.ParameterSetName -in ("Pipeline") -and ($UseDisplayNameAsFilename -or $UseIdAsFilename)) {
+                if ($UseDisplayNameAsFilename) {
+                    # Use the display name of the Analytics Rule as filename
+                    $NewFileName = $resource.properties.displayName -Replace '[^0-9A-Z]', ' '
+                    # Convert To CamelCase
+                    $NewFileName = ((Get-Culture).TextInfo.ToTitleCase($NewFileName) -Replace ' ') + '.yaml'
+                }
+                if ($UseIdAsFilename) {
+                    # Use id as of the Analytics Rule filename
+                    $NewFileName = $Id + '.yaml'
+                }
+                $OutFile = Join-Path -Path $Directory -ChildPath $NewFileName
             }
-            if ($UseDisplayNameAsFilename) {
-                # Use the display name of the Analytics Rule as filename
-                $NewFileName = $AnalyticsRuleTemplate.resources.properties.displayName -Replace '[^0-9A-Z]', ' '
-                # Convert To CamelCase
-                $NewFileName = ((Get-Culture).TextInfo.ToTitleCase($NewFileName) -Replace ' ') + '.yaml'
+            #endregion
+
+            # Get the properties of the analytic rule
+            $AnalyticsRule = $resource | Select-Object -ExpandProperty properties
+            # Add the id and kind from the ARM template
+            $AnalyticsRule = $AnalyticsRule | Add-Member -MemberType NoteProperty -Name "id" -Value $Id -PassThru -Force
+            $AnalyticsRule = $AnalyticsRule | Add-Member -MemberType NoteProperty -Name "kind" -Value $AnalyticsRuleTemplate.resources.kind -PassThru -Force
```

Comment:
Use `$resource.kind` instead of `$AnalyticsRuleTemplate.resources.kind` as this results in broken YAML files where all possible kind values are exported. Same for the check if a `templateVersion` exists

```powershell
            $AnalyticsRule = $AnalyticsRule | Add-Member -MemberType NoteProperty -Name "kind" -Value $resource.kind -PassThru -Force
            # Add version if not present
            if ( [string]::IsNullOrWhiteSpace($resource.properties.templateVersion) ) {
                $AnalyticsRule = $AnalyticsRule | Add-Member -MemberType NoteProperty -Name "version" -Value "1.0.0" -PassThru -Force
            }
```